Latex export

akmorrison opened the pull request:
It now exports full procedureSuites, which is nice.

It does not generate a full, compilable latex file by design, since there's a bunch of extra stuff we add to our ops files (operator macro defines, titles, etc) that aren't present in the proclang files.

Tests include generating a full procedureSuite export from `example.proc`, so I'm confident that it works for full procedures.

Not using an enum for `fmt` in export functions because neither `procedures.py` nor `conditions.py` import `topside`, which would be necessary for them to share an enum (I think), and I'm scared of adding that as an import since I don't know if it causes recursion problems.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/waterloo-rocketry/topside/90)
<!-- Reviewable:end -->

akmorrison committed:
Add latex export class function

akmorrison committed:
proc: Add export function to ProcedureSuite

akmorrison committed:
proc: Add conditions to latex export

akmorrison committed:
proc: Add latex export to all condition classes

Results in super weird looking procedures, but if you really wanna have
complicated wait conditions in your ops, that's on you.

akmorrison committed:
proc: Add proper NotImplementedError strings

akmorrison committed:
proc: fix failing test due to relative path

jacobdeery reviewed (commented):
Cool, looks pretty good! There shouldn't be any issue with importing `topside` in `procedures.py` and `conditions.py`.

 A few general formatting/style comments...

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-:-MJiqHkO0Qcsm29g87mM:b-b6594b)* status: 0 of 4 files reviewed, 13 unresolved discussions (waiting on @akmorrison and @jacobdeery)

___
*[topside/procedures/conditions.py, line 65 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJigDs2EbZiPWmKiSsi:-MJigDs31H92f8yti2Re:bu5i34c) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L65)):*
> ```
> \
> ```
I don't think you need to escape double quotes when the string is single-quoted
___
*[topside/procedures/conditions.py, line 271 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJigg410QEplLwZEa8X:-MJigg410QEplLwZEa8Y:bl0bc3g) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L271)):*
> ```
> return self.node + " is equal to " + str(round(self.reference_pressure)) + "psi"
> ```

You can use an f-string here too, which lets you drop the `str` and `+`

Also, thoughts on just using `==` instead of `is equal to`?
___
*[topside/procedures/conditions.py, line 285 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihBA63U4OjaaiyxyG:-MJihBA63U4OjaaiyxyH:bnj7hl0) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L285)):*
> ```Python
>     def export(self, fmt):
>         if fmt == 'latex':
>             return self.node + " is less than " + str(round(self.reference_pressure)) + "psi"
> ```

same as above
___
*[topside/procedures/conditions.py, line 299 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihDxZ4HyL4BILzgIz:-MJihDxZ4HyL4BILzgJ-:bnj7hl0) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L299)):*
> ```Python
>     def export(self, fmt):
>         if fmt == 'latex':
>             return self.node + " is greater than " + str(round(self.reference_pressure)) + "psi"
> ```

same as above
___
*[topside/procedures/conditions.py, line 313 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihFIj6xiUe_Rv8xnl:-MJihFIj6xiUe_Rv8xnm:bnj7hl0) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L313)):*
> ```Python
>     def export(self, fmt):
>         if fmt == 'latex':
>             return self.node + " is less than or equal to " + str(round(self.reference_pressure)) + "psi"
> ```

same as above
___
*[topside/procedures/conditions.py, line 327 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihGEhAECeUgec3WRI:-MJihGEhAECeUgec3WRJ:bnj7hl0) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L327)):*
> ```Python
>     def export(self, fmt):
>         if fmt == 'latex':
>             return self.node + " is greater than or equal to " + str(round(self.reference_pressure)) + "psi"
> ```

same as above
___
*[topside/procedures/procedure.py, line 26 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihJ8iDkPh2YtJL3XZ:-MJihJ8iDkPh2YtJL3X_:bgcui63) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/procedure.py#L26)):*
> ```
> "
> ```
single quotes throughout this function
___
*[topside/procedures/procedure.py, line 38 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJikRB_7aYDwQm92A29:-MJikRB_7aYDwQm92A2A:b6zxcfh) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/procedure.py#L38)):*
> ```
> class MiscAction:
> ```
This class needs an `export` function
___
*[topside/procedures/procedure.py, line 165 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJiiYpD3IMy4wmQj1IS:-MJiiYpD3IMy4wmQj1IT:b-w7rov3) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/procedure.py#L165)):*
> ```
> "
> ```
Single quotes throughout
___
*[topside/procedures/procedure.py, line 169 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJii4bL5c7R7ZaASH_e:-MJii4bL5c7R7ZaASH_f:b-h0zc2k) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/procedure.py#L169)):*
> ```
> i
> ```
using `i` here is kinda evil. How about `step`?
___
*[topside/procedures/procedure.py, line 174 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihsGe0D78ANQFzPPe:-MJihsGe0D78ANQFzPPf:b-6ur24j) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/procedure.py#L174)):*
> ```
> type(t[0]).__name__ != 'Immediate'
> ```
you can just do `not isinstance(t[0], top.Immediate)`
___
*[topside/procedures/tests/test_latex_export.py, line 3 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJij2p4-0p8R5Tu3sJC:-MJij2p4-0p8R5Tu3sJD:b-1cwozl) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/tests/test_latex_export.py#L3)):*
> ```
> import pytest
> import topside as top
> import os
> ```

Should be:

```
import os

import pytest

import topside as top
```

[PEP 8](https://www.python.org/dev/peps/pep-0008/#imports)
___
*[topside/procedures/tests/test_latex_export.py, line 19 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJijIph1R-eAvo_f8r0:-MJijIph1R-eAvo_f8r1:b6t8bqj) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/tests/test_latex_export.py#L19)):*
<details><summary><i alt="summary">Quoted 5 lines of code…</i></summary>

> ```
>     expected_export = r'''\subsection{p1}
> \begin{checklist}
>     \item \PRIMARY{} Open remote fill valve
>     \item \SECONDARY{} Close remote vent valve
> \end{checklist}'''
> ```
</details>

if you use `textwrap.dedent` this string can be indented as normal


<!-- Sent from Reviewable.io -->

akmorrison committed:
proc: Review fixes for LaTeX export

Changes in this commit:
        - Use an enum for export format
        - Replace double quotes with single quotes
        - Don't escape double quotes in strings
        - Use f strings instead of a lot of string addition
        - Add export function to MiscAction class
        - Replace i as iteration variable
        - Use PEP8 import line break style
        - Use textwrap.dedent for multi-line strings

akmorrison reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/topside/90)* status: 0 of 4 files reviewed, 13 unresolved discussions (waiting on @akmorrison and @jacobdeery)

___
*[topside/procedures/conditions.py, line 65 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJigDs2EbZiPWmKiSsi:-MK6lKfoBIsOAbwZzHDL:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L65)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> \
> ```
I don't think you need to escape double quotes when the string is single-quoted
</blockquote></details>

Done.
___
*[topside/procedures/conditions.py, line 271 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJigg410QEplLwZEa8X:-MK6lasxCd7Go1TypwdK:b2wkivv) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L271)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> return self.node + " is equal to " + str(round(self.reference_pressure)) + "psi"
> ```

You can use an f-string here too, which lets you drop the `str` and `+`

Also, thoughts on just using `==` instead of `is equal to`?
</blockquote></details>

I prefer `is equal to` (and all the other spelled out conditions) because they're wordier/harder to misread when printed on paper. I agree they make big conditions way worse, but I don't think we expect to ever have more than two conditions in a printed procedure.

Agree on the f-string thing though, will do (and use single quotes) 
___
*[topside/procedures/conditions.py, line 285 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihBA63U4OjaaiyxyG:-MK6mZMV0NiEGJXFTioY:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L285)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

same as above
</blockquote></details>

Done.
___
*[topside/procedures/conditions.py, line 299 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihDxZ4HyL4BILzgIz:-MK6m_w7DCtChJurCjE0:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L299)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

same as above
</blockquote></details>

Done.
___
*[topside/procedures/conditions.py, line 313 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihFIj6xiUe_Rv8xnl:-MK6mbTM50U_ffSfvCu8:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L313)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

same as above
</blockquote></details>

Done.
___
*[topside/procedures/conditions.py, line 327 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihGEhAECeUgec3WRI:-MK6mbruCaJUMR-5te-w:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L327)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

same as above
</blockquote></details>

Done.
___
*[topside/procedures/procedure.py, line 38 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJikRB_7aYDwQm92A29:-MK6nfw551FMGTMTLjH7:b-at5v50) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/procedure.py#L38)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> class MiscAction:
> ```
This class needs an `export` function
</blockquote></details>

I added one to  return action_type 
___
*[topside/procedures/procedure.py, line 165 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJiiYpD3IMy4wmQj1IS:-MK6n9iV0_OG5pwfxyK9:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/procedure.py#L165)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> "
> ```
Single quotes throughout
</blockquote></details>

Done.
___
*[topside/procedures/procedure.py, line 169 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJii4bL5c7R7ZaASH_e:-MK6nIGACd3QEQAOjXEY:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/procedure.py#L169)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> i
> ```
using `i` here is kinda evil. How about `step`?
</blockquote></details>

Done.
___
*[topside/procedures/procedure.py, line 174 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihsGe0D78ANQFzPPe:-MK6nV2XENWIiIbeNj-B:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/procedure.py#L174)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> type(t[0]).__name__ != 'Immediate'
> ```
you can just do `not isinstance(t[0], top.Immediate)`
</blockquote></details>

Done.
___
*[topside/procedures/tests/test_latex_export.py, line 3 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJij2p4-0p8R5Tu3sJC:-MK6q25U9IT2qYZ-_z4A:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/tests/test_latex_export.py#L3)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> import pytest
> import topside as top
> import os
> ```

Should be:

```
import os

import pytest

import topside as top
```

[PEP 8](https://www.python.org/dev/peps/pep-0008/#imports)
</blockquote></details>

Done.
___
*[topside/procedures/tests/test_latex_export.py, line 19 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJijIph1R-eAvo_f8r0:-MK6q2bj0GkXI4oAhYHT:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/tests/test_latex_export.py#L19)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
>     expected_export = r'''\subsection{p1}
> \begin{checklist}
>     \item \PRIMARY{} Open remote fill valve
>     \item \SECONDARY{} Close remote vent valve
> \end{checklist}'''
> ```
if you use `textwrap.dedent` this string can be indented as normal
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

akmorrison reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/topside/90)* status: 0 of 4 files reviewed, 13 unresolved discussions (waiting on @jacobdeery)

___
*[topside/procedures/procedure.py, line 26 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJihJ8iDkPh2YtJL3XZ:-MK6sRO89RljO0R-sC20:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/procedure.py#L26)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> "
> ```
single quotes throughout this function
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

jacobdeery reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/topside/90)* status: 0 of 4 files reviewed, 3 unresolved discussions (waiting on @akmorrison)

___
*[topside/procedures/conditions.py, line 271 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJigg410QEplLwZEa8X:-MK6uG87Cxds-tBybJFP:b-b6bd3j) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/conditions.py#L271)):*
<details><summary><i>Previously, akmorrison (Aaron Morrison) wrote…</i></summary><blockquote>

I prefer `is equal to` (and all the other spelled out conditions) because they're wordier/harder to misread when printed on paper. I agree they make big conditions way worse, but I don't think we expect to ever have more than two conditions in a printed procedure.

Agree on the f-string thing though, will do (and use single quotes) 
</blockquote></details>

Cool, makes sense
___
*[topside/procedures/procedure.py, line 38 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJikRB_7aYDwQm92A29:-MK6uV5r7oJtb5tq7Ym_:bsrfkwe) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/procedure.py#L38)):*
<details><summary><i>Previously, akmorrison (Aaron Morrison) wrote…</i></summary><blockquote>

I added one to  return action_type 
</blockquote></details>

Looks good
___
*[topside/procedures/tests/test_latex_export.py, line 3 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJij2p4-0p8R5Tu3sJC:-MK6uk2f28g-9J2I3xmr:b-yn8r60) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/tests/test_latex_export.py#L3)):*
<details><summary><i>Previously, akmorrison (Aaron Morrison) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Super-nit: `textwrap` is a standard library import, so it should be grouped with `os`
___
*[topside/procedures/tests/test_latex_export.py, line 19 at r1](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MJijIph1R-eAvo_f8r0:-MK6xkEg7U1jwq68iJjG:b-9stdbe) ([raw file](https://github.com/waterloo-rocketry/topside/blob/ebccae457d42265cb3d57cd35e971b4dab088d19/topside/procedures/tests/test_latex_export.py#L19)):*
<details><summary><i>Previously, akmorrison (Aaron Morrison) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

hmm, I didn't realize it would look like that with a raw string. That's a bit annoying (but fine for now)
___
*[topside/procedures/tests/test_latex_export.py, line 65 at r2](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MK6vVwC6UfMyjp1EM89:-MK6vVwC6UfMyjp1EM8A:byp18fd) ([raw file](https://github.com/waterloo-rocketry/topside/blob/203ce664553cae54d76bb2f22f699d3d83397f38/topside/procedures/tests/test_latex_export.py#L65)):*
> ```Python
>     export = suite.export(top.ExportFormat.Latex)
> 
>     print(export)
> ```

Remove
___
*[topside/procedures/tests/test_latex_export.py, line 105 at r2](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MK6vUpH4m9H7POC5RlJ:-MK6vUpH4m9H7POC5RlK:byp18fd) ([raw file](https://github.com/waterloo-rocketry/topside/blob/203ce664553cae54d76bb2f22f699d3d83397f38/topside/procedures/tests/test_latex_export.py#L105)):*
> ```Python
>     export = suite.export(top.ExportFormat.Latex)
> 
>     print(export)
> ```

Remove


<!-- Sent from Reviewable.io -->

akmorrison committed:
proc: Review fixes

Changes in this commit:
        - Put textwrap with the standard library imports
        - Remove print statements in test

akmorrison reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/topside/90)* status: 0 of 4 files reviewed, 3 unresolved discussions (waiting on @akmorrison and @jacobdeery)

___
*[topside/procedures/tests/test_latex_export.py, line 65 at r2](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MK6vVwC6UfMyjp1EM89:-MKHERXb8pKu9aWZWSAl:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/203ce664553cae54d76bb2f22f699d3d83397f38/topside/procedures/tests/test_latex_export.py#L65)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

Remove
</blockquote></details>

Done.
___
*[topside/procedures/tests/test_latex_export.py, line 105 at r2](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-MK6vUpH4m9H7POC5RlJ:-MKHESCo5IiI1KBq01qD:b-896fix) ([raw file](https://github.com/waterloo-rocketry/topside/blob/203ce664553cae54d76bb2f22f699d3d83397f38/topside/procedures/tests/test_latex_export.py#L105)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

Remove
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

jacobdeery reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, thanks for taking this on!

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/topside/90#-:-MKHEfvkD7eszOkjxdg2:bx2kd3n)* status: 0 of 4 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->
